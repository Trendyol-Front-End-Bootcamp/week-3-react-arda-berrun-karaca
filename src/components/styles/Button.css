.btn {
    width: 100px;
    border: none;
    outline: none;
    padding: 10px 14px;
    border-radius: 5px;
    background-color: var(--dark-green);
    color: var(--main-white);
    /* margin-right: .5rem; */
    cursor: pointer;
    font-size: 1.125rem;
  }

