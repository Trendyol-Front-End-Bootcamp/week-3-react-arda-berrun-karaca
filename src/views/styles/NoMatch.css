.nomatch {
  width: 80%;
  height: 100vh;
  display: grid;
  place-items: center;
  column-gap: 1rem;
  text-align: center;
  margin: 0 auto;
}

.nomatch__container h1 {
  font-size: 8rem;
  color: var(--dark-green);
}

.nomatch__container h3 {
  color: var(--main-white);
}

.nomatch__container h4 {
  font-size: 1.5rem;
  color: var(--dark-green);
}


@media screen and (min-width: 576px) {
  .nomatch {
    max-width: var(--container);
  }
  .nomatch__container h1 {
    font-size: 12rem;
  }
  .nomatch__container h3 {
    font-size: 2rem;
  }
}

@media screen and (min-width: 768px) {
  .nomatch__container h1 {
    font-size: 15rem;
  }
  .nomatch__container h3 {
    font-size: 2.5rem;
  }
}
