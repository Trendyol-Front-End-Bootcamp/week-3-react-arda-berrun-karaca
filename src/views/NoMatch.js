import React from 'react';
import { useHistory } from 'react-router-dom';
import Button from '../components/Button';
import './styles/NoMatch.css';

function NoMatch() {
  const history = useHistory();

  return (
    <div className="nomatch">
      <div className="nomatch__container">
        <h3>Oh no! Seems you're in a pickle, Rick!</h3>
        <h4>Page not found</h4>
        <h1>404</h1>
        <Button onClick={() => history.push('/')}>Go Back</Button>
      </div>
    </div>
  );
}

export default NoMatch;
