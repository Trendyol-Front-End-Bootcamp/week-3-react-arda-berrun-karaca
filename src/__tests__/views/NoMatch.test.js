import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { createMemoryHistory } from 'history';
import { Router, BrowserRouter } from 'react-router-dom';

import NoMatch from '../../views/NoMatch';
import Button from '../../components/Button';

describe('NoMatch tests', () => {
  it('should be defined', () => {
    expect(NoMatch).toBeDefined();
  });

  it('should render correctly', () => {
    const wrapper = renderer
      .create(
        <BrowserRouter>
          <NoMatch />
        </BrowserRouter>
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should push to the correct location when the button clicked', () => {
    const history = createMemoryHistory();

    const wrapper = mount(
      <Router history={history}>
        <NoMatch />
      </Router>
    );

    wrapper.find(Button).simulate('click');

    expect(history.location.pathname).toBe('/');
  });
});
