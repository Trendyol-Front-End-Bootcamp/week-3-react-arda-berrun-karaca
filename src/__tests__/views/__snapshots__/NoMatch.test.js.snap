// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch tests should render correctly 1`] = `
<div
  className="nomatch"
>
  <div
    className="nomatch__container"
  >
    <h3>
      Oh no! Seems you're in a pickle, Rick!
    </h3>
    <h4>
      Page not found
    </h4>
    <h1>
      404
    </h1>
    <button
      className="btn"
      onClick={[Function]}
    >
      Go Back
    </button>
  </div>
</div>
`;
